### PR TITLE
Resolve deployment failure that is caused by insufficient memory.

### PR DESCRIPTION
**Pull request by galiacheng**
This pull request includes updates to resolve deployment failure that is caused by insufficient memory.

Here are the most significant changes:

Version Update:

* [`pom.xml`](diffhunk://#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8L43-R43): The version of the `wls-on-aks-azure-marketplace` has been updated from `1.0.76` to `1.0.78`.

User Interface Improvements:

* [`weblogic-azure-aks/src/main/arm/createUiDefinition.json`](diffhunk://#diff-6942d0c3ae64307ad19b9c3cb12dbb240644dd47aefa4b3243c182ef0ec633acR21-R28): Added a new section in the UI for selecting VM sizes, and updated the sliders for minimum and maximum node count to use dynamic values based on the selected VM size and the WebLogic cluster size. Also, a section for node size selection has been moved and expanded with more constraints and options. [[1]](diffhunk://#diff-6942d0c3ae64307ad19b9c3cb12dbb240644dd47aefa4b3243c182ef0ec633acR21-R28) [[2]](diffhunk://#diff-6942d0c3ae64307ad19b9c3cb12dbb240644dd47aefa4b3243c182ef0ec633acR323-R396) [[3]](diffhunk://#diff-6942d0c3ae64307ad19b9c3cb12dbb240644dd47aefa4b3243c182ef0ec633acL332-L360)

Resource Validation:

* [`weblogic-azure-aks/src/main/arm/scripts/inline-scripts/validateParameters.sh`](diffhunk://#diff-b94e7f0c9e182e988b6d1e7254279af3c1b07f5aa63bbb39c2fd78b3a21b5defR129-R151): Added a new function `validate_memory_resources` to check if the AKS cluster has sufficient memory resources based on the number of application replicas. The function is then called in the main script. [[1]](diffhunk://#diff-b94e7f0c9e182e988b6d1e7254279af3c1b07f5aa63bbb39c2fd78b3a21b5defR129-R151) [[2]](diffhunk://#diff-b94e7f0c9e182e988b6d1e7254279af3c1b07f5aa63bbb39c2fd78b3a21b5defR627-R628)
* `weblogic-azure-aks/src/main/bicep/mainTemplate.bicep`, `weblogic-azure-aks/src/main/bicep/modules/_deployment-scripts/_ds-validate-parameters.bicep`: Passed the number of application replicas as a parameter to the validation script. [[1]](diffhunk://#diff-9eaec954615fbe14ad7f6da8d53b31242e9a213bf900eb0f172998960b43c54cR412) [[2]](diffhunk://#diff-c15f35a79ccd16ec8c570aa05b668e8725a5823678c6867d1cc2a9d17e36b04dR15) [[3]](diffhunk://#diff-c15f35a79ccd16ec8c570aa05b668e8725a5823678c6867d1cc2a9d17e36b04dR125-R128)

### CODE REVIEW

File: weblogic-azure-aks/src/main/arm/createUiDefinition.json [hunk 1]
```diff
@@ -311,14 +319,81 @@
...
+                                ],
+                                "constraints": {
+                                    "numAvailabilityZonesRequired": 3,
+                                    "zone": "3",
+                                    "excludedSizes": [
+                                        "Standard_A0",
+                                        "Standard_A1",
+                                        "Standard_A1_v2",
+                                        "Standard_F1",
+                                        "Standard_F1s",
```
Comment (galiacheng) [inline weblogic-azure-aks/src/main/arm/createUiDefinition.json:339]: Those are from https://aka.ms/aks/restricted-skus.

File: weblogic-azure-aks/src/main/arm/createUiDefinition.json [hunk 2]
```diff
@@ -18,6 +18,14 @@
                 "type": "Microsoft.Common.Section",
                 "label": "Credentials for WebLogic",
                 "elements": [
+                    {
+                        "name": "listVMSizes",
```
Comment (galiacheng) [inline weblogic-azure-aks/src/main/arm/createUiDefinition.json:22]: Move the REST API query to the **Basics** blade, then the data may be ready when comes to **AKS** blade.

File: weblogic-azure-aks/src/main/arm/createUiDefinition.json [hunk 3]
```diff
@@ -311,14 +319,81 @@
...
+                            },
                             {
                                 "name": "aksNodeCount",
                                 "type": "Microsoft.Common.Slider",
-                                "min": 1,
-                                "max": 100,
-                                "label": "Minimum node count",
+                                "min": "[add(1, div(add(12288, mul(if(empty(basics('basicsOptional').wlsClusterSize),5,basics('basicsOptional').wlsClusterSize), 1536)), first(filter(basics('basicsRequired').listVMSizes.value, (item) => equals(item.name, steps('section_aks').clusterInfo.nodeVMSizeSelector))).memoryInMB))]",
                                 "defaultValue": 3,
+                                "max": 1000,
```
Comment (galiacheng) [inline weblogic-azure-aks/src/main/arm/createUiDefinition.json:395]: The same max value with AKS service.

**Comment by galiacheng** [comment]
Have problem to work with `Standard_A2_v2`.

**Comment by galiacheng** [comment]
Hello @edburns this PR is ready for review, including `livenessProbe` and `readinessProbe` tuning for A-series VM sizes.

**Review by edburns — APPROVED** [review]

**Review by mriccell — APPROVED** [review]

**Pull request merged**